src/agent/Core/ApplicationPool/Group/InternalUtils.cpp

```diff
@@ -325,6 +335,33 @@ Group::callAbortLongRunningConnectionsCallback(const ProcessPtr &process) {
 	}
 }
 
+void
+Group::timeoutRequestsCallback() {
+	boost::shared_ptr<Group> extraReferenceToMe = shared_from_this();
+	if (options.maxRequestQueueTime == 0) {
```

Comment:
This is not safe. You aren't holding the lock here.

Also, we want to support the case where `maxRequestQueueTime` was initially 0, but is changed later by the administrator to a non-zero value. The standard way to deal with this is by sleeping on a condition variable. Then, when any options are changed, the code can wake up the timeoutRequestsCallback thread by waking up the condition variable.